codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ManifestWriter.java

```diff
@@ -0,0 +1,157 @@
+                .max(Comparator.comparing(SymbolDependency::getVersion));
+
+        Map<StringNode, Node> manifestNodes = new HashMap<>();
+
+        Map<String, String> minimumDependencies = gatherMinimumDependencies(nonStdLib);
+
+        Map<StringNode, Node> dependencyNodes = new HashMap<>();
+        for (Map.Entry<String, String> entry : minimumDependencies.entrySet()) {
+            dependencyNodes.put(StringNode.from(entry.getKey()),
+                    StringNode.from(entry.getValue()));
+        }
+
+        Collection<String> generatedFiles = new ArrayList<>();
+        Path baseDir = fileManifest.getBaseDir();
+        for (Path filePath : fileManifest.getFiles()) {
+            generatedFiles.add(baseDir.relativize(filePath).toString());
+        }
+
+        manifestNodes.put(StringNode.from("module"), StringNode.from(settings.getModuleName()));
+        minStandard.ifPresent(symbolDependency ->
+                manifestNodes.put(StringNode.from("go"), StringNode.from(symbolDependency.getVersion())));
+        manifestNodes.put(StringNode.from("dependencies"), Node.objectNode(dependencyNodes));
+        manifestNodes.put(StringNode.from("files"), ArrayNode.fromStrings(generatedFiles));
+
+        return ObjectNode.objectNode(manifestNodes);
+    }
+
+    private static Map<String, String> gatherMinimumDependencies(
+            Stream<SymbolDependency> symbolStream
+    ) {
+        return SymbolDependency.gatherDependencies(symbolStream, (dx, dy) -> {
+            SemanticVersion sx = SemanticVersion.parseVersion(dx.getVersion());
+            SemanticVersion sy = SemanticVersion.parseVersion(dx.getVersion());
+
+            // This *shouldn't* happen in Go since the Go module import path must end with the major version component.
+            // Exception is the case where the major version is 0 or 1.
+            if (sx.getMajor() != sy.getMajor() && !(sx.getMajor() == 0 || sy.getMajor() == 0)) {
+                throw new CodegenException(String.format("Dependency %s has conflicting major versions",
+                        dx.getPackageName()));
+            }
+
+            int cmp = sx.compareTo(sy, ManifestWriter::compareGoPseudoVersions);
+            if (cmp < 0) {
+                return dy;
+            } else if (cmp > 0) {
+                return dx;
+            }
+            return dx;
+        }).entrySet().stream()
+                .flatMap(entry -> entry.getValue().entrySet().stream())
+                .collect(Collectors.toMap(
+                        Map.Entry::getKey, entry -> entry.getValue().getVersion(), (a, b) -> b, TreeMap::new));
+    }
+
+    private static int compareGoPseudoVersions(String x, String y) {
+        if (x.equals(INCOMPATIBLE_VERSION) && y.equals(INCOMPATIBLE_VERSION)) {
+            return 0;
+        }
+        return parsePseudoVersions(x).compareTo(parsePseudoVersions(y));
+    }
+
+    private static String parsePseudoVersions(String extra) {
```

Comment:
Could you put a comment here, to help understand the format of PseudoVersion this function is parsing.